components/bl-alert-component/src/index.js

```diff
@@ -0,0 +1,85 @@
+import { useState, useEffect } from 'react'
+import { AlertButton, AlertIcon, AlertTitle } from './subcomponents'
+
+export default function MyCustomComponent({ component, eventHandlers }) {
+  const {
+    display,
+    classList,
+    iconVisibility,
+    messageType,
+    messageTitle,
+    messageText,
+    variant,
+    closeButtonVisibility,
+    closingDuration,
+  } = component
+  const { onCloseButton } = eventHandlers
+
+  const [isAlertVisible, setIsAlertVisible] = useState(true)
+  const [isClosing, setIsClosing] = useState(false)
+  const classes = useAlertClasses(variant, messageType, isClosing)
+  const classesTitle = useTextClass('alert__title', variant, messageType)
+  const classesAlertText = useTextClass('alert__text', variant, messageType)
+
+  useEffect(() => {
+    setIsClosing(isAlertVisible)
```

Comment:
it's not clear for me, why do set `setIsClosing(true)` if the alert is visible 

```
useEffect(() => {
    setIsClosing(isAlertVisible)
  }, [isAlertVisible])
```

and then set it `false` when actually it's going to be closed 

```
 component.closeButtonAction = () => {
    setIsClosing(false)

  ...
```